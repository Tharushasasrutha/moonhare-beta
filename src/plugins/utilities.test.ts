import Processor from '..';
import CleanCSS from 'clean-css';

const cleanCss = new CleanCSS({format: 'beautify'});

const processor = new Processor();
function css(className: string, proc = processor) {
    return cleanCss.minify(proc.interpret(className).styleSheet.build()).styles;
}

describe('utilities', () => {
    test('dynamic', () => {
        expect([...processor.dynamicPlugins.keys()]).toMatchSnapshot();
    });

    test('container', () => {
        expect(css('container')).toMatchSnapshot();
    });

    test('container with padding', () => {
        expect(
            css(
                'container',
                new Processor({
                    theme: {
                        container: {padding: '2rem'}
                    }
                })
            )
        ).toMatchSnapshot();
        expect(
            css(
                'container',
                new Processor({
                    theme: {
                        container: {padding: {DEFAULT: '4rem'}}
                    }
                })
            )
        ).toMatchSnapshot();
    });

    test('container with center', () => {
        expect(
            css(
                'container',
                new Processor({
                    theme: {
                        container: {center: true}
                    }
                })
            )
        ).toMatchSnapshot();
    });

    test('container with custom breakpoints', () => {
        expect(
            css(
                'container',
                new Processor({
                    theme: {
                        container: {
                            screens: {
                                mobile: '768px',
                                desktop: '1280px'
                            }
                        }
                    }
                })
            )
        ).toMatchSnapshot();
    });

    test('container with custom breakpoints and padding', () => {
        expect(
            css(
                'container',
                new Processor({
                    theme: {
                        container: {
                            padding: {
                                DEFAULT: '4rem',
                                mobile: '2rem',
                                desktop: '8rem'
                            },
                            screens: {
                                mobile: '768px',
                                desktop: '1280px'
                            }
                        }
                    }
                })
            )
        ).toMatchSnapshot();
    });

    test('box-decoration-break', () => {
        expect(css('decoration-slice decoration-clone')).toMatchSnapshot();
    });

    test('box-sizing', () => {
        expect(css('box-content box-border')).toMatchSnapshot();
    });

    test('display', () => {
        expect(
            css(
                [
                    'block',
                    'inline-block',
                    'inline',
                    'flex',
                    'inline-flex',
                    'table',
                    'inline-table',
                    'table-caption',
                    'table-cell',
                    'table-column',
                    'table-column-group',
                    'table-footer-group',
                    'table-header-group',
                    'table-row-group',
                    'table-row',
                    'flow-root',
                    'grid',
                    'inline-grid',
                    'contents',
                    'list-item',
                    'hidden'
                ]
                    .reverse()
                    .join(' ')
            )
        ).toMatchSnapshot();
    });

    test('float', () => {
        expect(css('float-right float-none float-left')).toMatchSnapshot();
    });

    test('clear', () => {
        expect(
            css('clear-right clear-none clear-both clear-left')
        ).toMatchSnapshot();
    });

    test('isolation', () => {
        expect(css('isolate isolation-auto')).toMatchSnapshot();
    });

    test('object-fit', () => {
        expect(
            css(
                'object-contain object-cover object-fill object-none object-scale-down'
            )
        ).toMatchSnapshot();
    });

    test('object-position', () => {
        expect(
            css(
                'object-bottom object-center object-left object-right object-top object-left-top object-left-bottom object-right-top object-right-bottom'
            )
        ).toMatchSnapshot();
    });

    test.todo(
        'object-position with config' /*, () => {
        expect(
            css(
                '',
                new Processor({
                    theme: {
                        objectPosition: {}
                    }
                })
            )
        ).toMatchSnapshot();
    }*/
    );

    test('grid-cols', () => {
        expect(css('grid-cols-10')).toMatchSnapshot();
    });

    test('height', () => {
        expect(css('h-7')).toMatchSnapshot();
    });
});
