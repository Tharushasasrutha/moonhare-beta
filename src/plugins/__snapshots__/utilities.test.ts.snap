// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utilities box-decoration-break 1`] = `
".decoration-slice {
  box-decoration-break: slice
}
.decoration-clone {
  box-decoration-break: clone
}"
`;

exports[`utilities box-sizing 1`] = `
".box-border {
  box-sizing: border-box
}
.box-content {
  box-sizing: content-box
}"
`;

exports[`utilities clear 1`] = `
".clear-left {
  clear: left
}
.clear-right {
  clear: right
}
.clear-both {
  clear: both
}
.clear-none {
  clear: none
}"
`;

exports[`utilities container 1`] = `
".container {
  width: 100%
}
@media (min-width:640px) {
  .container {
    width: 640px
  }
}
@media (min-width:768px) {
  .container {
    width: 768px
  }
}
@media (min-width:1024px) {
  .container {
    width: 1024px
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px
  }
}
@media (min-width:1536px) {
  .container {
    width: 1536px
  }
}"
`;

exports[`utilities container with center 1`] = `
".container {
  width: 100%;
  margin-right: auto;
  margin-left: auto
}
@media (min-width:640px) {
  .container {
    width: 640px
  }
}
@media (min-width:768px) {
  .container {
    width: 768px
  }
}
@media (min-width:1024px) {
  .container {
    width: 1024px
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px
  }
}
@media (min-width:1536px) {
  .container {
    width: 1536px
  }
}"
`;

exports[`utilities container with custom breakpoints 1`] = `
".container {
  width: 100%
}
@media (min-width:768px) {
  .container {
    width: 768px
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px
  }
}"
`;

exports[`utilities container with custom breakpoints and padding 1`] = `
".container {
  width: 100%;
  padding-left: 4rem;
  padding-right: 4rem
}
@media (min-width:768px) {
  .container {
    width: 768px;
    padding-left: 2rem;
    padding-right: 2rem
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px;
    padding-left: 8rem;
    padding-right: 8rem
  }
}"
`;

exports[`utilities container with padding 1`] = `
".container {
  width: 100%;
  padding-left: 2rem;
  padding-right: 2rem
}
@media (min-width:640px) {
  .container {
    width: 640px
  }
}
@media (min-width:768px) {
  .container {
    width: 768px
  }
}
@media (min-width:1024px) {
  .container {
    width: 1024px
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px
  }
}
@media (min-width:1536px) {
  .container {
    width: 1536px
  }
}"
`;

exports[`utilities container with padding 2`] = `
".container {
  width: 100%;
  padding-left: 4rem;
  padding-right: 4rem
}
@media (min-width:640px) {
  .container {
    width: 640px
  }
}
@media (min-width:768px) {
  .container {
    width: 768px
  }
}
@media (min-width:1024px) {
  .container {
    width: 1024px
  }
}
@media (min-width:1280px) {
  .container {
    width: 1280px
  }
}
@media (min-width:1536px) {
  .container {
    width: 1536px
  }
}"
`;

exports[`utilities display 1`] = `
".block {
  display: block
}
.inline-block {
  display: inline-block
}
.inline {
  display: inline
}
.flex {
  display: flex
}
.inline-flex {
  display: inline-flex
}
.table {
  display: table
}
.inline-table {
  display: inline-table
}
.table-caption {
  display: table-caption
}
.table-cell {
  display: table-cell
}
.table-column {
  display: table-column
}
.table-column-group {
  display: table-column-group
}
.table-footer-group {
  display: table-footer-group
}
.table-header-group {
  display: table-header-group
}
.table-row-group {
  display: table-row-group
}
.table-row {
  display: table-row
}
.flow-root {
  display: flow-root
}
.grid {
  display: grid
}
.inline-grid {
  display: inline-grid
}
.contents {
  display: contents
}
.list-item {
  display: list-item
}
.hidden {
  display: none
}"
`;

exports[`utilities dynamic 1`] = `
Array [
  "object",
  "top|right|bottom|left|inset(-x|-y)?",
  "z",
  "bg",
  "h|w",
  "space-(x|y)",
  "grid-(cols|rows)",
]
`;

exports[`utilities float 1`] = `
".float-right {
  float: right
}
.float-left {
  float: left
}
.float-none {
  float: none
}"
`;

exports[`utilities grid-cols 1`] = `
".grid-cols-10 {
  grid-template-columns: repeat(10,minmax(0,1fr))
}"
`;

exports[`utilities height 1`] = `
".h-7 {
  height: 1.75rem
}"
`;

exports[`utilities isolation 1`] = `
".isolate {
  isolation: isolate
}
.isolation-auto {
  isolation: auto
}"
`;

exports[`utilities object-fit 1`] = `
".object-contain {
  object-position: contain
}
.object-cover {
  object-position: cover
}
.object-fill {
  object-position: fill
}
.object-none {
  object-position: none
}
.object-scale-down {
  object-position: scale down
}"
`;

exports[`utilities object-position 1`] = `
".object-bottom {
  object-position: bottom
}
.object-center {
  object-position: center
}
.object-left {
  object-position: left
}
.object-right {
  object-position: right
}
.object-top {
  object-position: top
}
.object-left-top {
  object-position: left top
}
.object-left-bottom {
  object-position: left bottom
}
.object-right-top {
  object-position: right top
}
.object-right-bottom {
  object-position: right bottom
}"
`;
